tests/test_main_github.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+import os
+import pytest
+from urlchecker.main.github import clone_repo, delete_repo, get_branch
+
+
+@pytest.mark.parametrize('git_path', ["https://github.com/urlstechie/urlchecker-test-repo"])
+def test_clone_and_del_repo(git_path):
+    """
+    test clone and del repo function.
+    """
+    # del repo if it exisits
+    if os.path.exists(os.path.basename(git_path)):
+        delete_repo(os.path.basename(git_path))
+
+    # clone
+    base_path = clone_repo(git_path)
+    assert(base_path == os.path.basename(git_path))
+
+    # delete should have return code of 0 (success)
+    if not delete_repo(base_path) == 0:
+        raise AssertionError
+
+
+def test_get_branch():
+    """
+    test getting branch from environment or default
+    """
+    # Unset defaults to master
+    branch = get_branch()
+    if branch != "master":
+        raise AssertionError
+
+    # Set both GitHub input variable and ref (ref takes priority)
+    for pair in [["INPUT_BRANCH", "devel"], ["GITHUB_REF", "refs/heads/branchy"]]:
+        os.environ[pair[0]] = pair[1]
+        os.putenv(pair[0], pair[1])
+
+    # Second preference should be for INPUT_BRANCH
+    branch = get_branch()
+    if branch != "devel":
+        raise AssertionError
+
+    del os.environ['INPUT_BRANCH']
```

Comment:
I'm fairly sure that I wrote this test, but why is it listed as a new file with your name? If you do a rename of tests, it would be best to do that first, otherwise I lose all the credit for my previous work :/

Reply:
This is definitely your test not mine but I moved a lot of stuff around so that might be the cause of it.
Unfortunately, I am not sure how to handle this situation (loss of previous work credits). Any suggestions?

Reply:
Normally when you move a file, GitHub will notice that it's just a rename (and call it that) but it looks like what you did here is to copy paste content to a new file. What you can do is open a separate PR that instead copies a file (for the rename) and then tweaks any content, if needed. I would recommend having the repo cloned locally twice - one with changes from this branch, and another with a fresh branch where you can start adding the changes. You can try doing a test with just one file if it helps.

Reply:
We could also try https://help.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors#creating-co-authored-commits-on-the-command-line but that might not work. It would be good if we talked about how you went about creating this PR so we know maybe what you could do differently for next time.